src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -90,7 +95,17 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
 
 
             var containerType = context.Key.ContainerType ?? context.Key.ModelType;
-            var localizer = _stringLocalizerFactory?.Create(containerType);
+            IStringLocalizer localizer;
+
+            if (_stringLocalizerFactory != null && _options?.Value?.DataAnnotationLocalizerProvider != null)
+            {
+                localizer = _options.Value.DataAnnotationLocalizerProvider(containerType, _stringLocalizerFactory);
+            }
+            else
+            {
+                localizer = _stringLocalizerFactory?.Create(containerType);
```

Comment:
I guess I was trying to give it a default in case DataAnnotationLocalizerProvider wasn't set, but now that you point it out, since that's the only thing on `MvcDataAnnotationsLocalizationOptions` that would be pretty silly, making this a needless complication. Will remove.